src/screens/UserPortal/Settings/Settings.tsx

```diff
@@ -21,148 +24,515 @@ export default function settings(): JSX.Element {
+                        </Button>
+                        <Form.Control
+                          accept="image/*"
+                          id="postphoto"
+                          name="photo"
+                          type="file"
+                          className={styles.cardControl}
+                          data-testid="fileInput"
+                          multiple={false}
+                          ref={fileInputRef}
+                          onChange={
+                            /* istanbul ignore next */
+                            async (
+                              e: React.ChangeEvent<HTMLInputElement>,
+                            ): Promise<void> => {
+                              const target = e.target as HTMLInputElement;
+                              const file = target.files && target.files[0];
+                              if (file) {
+                                const image = await convertToBase64(file);
+                                setUserDetails({ ...userDetails, image });
+                              }
+                            }
+                          }
+                          style={{ display: 'none' }}
+                        />
+                      </div>
+                    </Col>
+                  </Row>
+                  <Row className="mb-1">
+                    <Col lg={4}>
+                      <Form.Label
+                        htmlFor="birthDate"
+                        className={`${styles.cardLabel}`}
+                      >
+                        {t('birthDate')}
+                      </Form.Label>
+                      <Form.Control
+                        type="date"
+                        id="birthDate"
+                        value={userDetails.birthDate}
+                        onChange={(e) =>
+                          handleFieldChange('birthDate', e.target.value)
+                        }
+                        className={`${styles.cardControl}`}
+                      />
+                    </Col>
+                    <Col lg={4}>
+                      <Form.Label
+                        htmlFor="grade"
+                        className={`${styles.cardLabel}`}
+                      >
+                        {t('grade')}
+                      </Form.Label>
+                      <Form.Control
+                        as="select"
+                        id="grade"
+                        value={userDetails.grade}
+                        onChange={(e) =>
+                          handleFieldChange('grade', e.target.value)
+                        }
+                        className={`${styles.cardControl}`}
+                        data-testid="inputGrade"
+                      >
+                        <option value="" disabled>
+                          {t('gradePlaceholder')}
+                        </option>
+                        <option value="GRADE_1">{t('grade1')}</option>
```

Comment:
Using this approach of embedding all the options directly within the code for the dropdown menu will become increasingly challenging to maintain in the future and is generally considered poor practice.

Reply:
I've taken a different approach for handling this issue. I created a file in the utilities directory that contains all the enums for each field, along with their corresponding labels and values. These enums are then directly imported into the code, providing a more organized and manageable solution.

Reference: https://github.com/PalisadoesFoundation/talawa-admin/pull/1667/files#diff-2d7b9c41af3deb393284274ccf52ffe702052af5e80d87ed83e9c37606af98cd

Reply:
ok, I will place the values as `gradeOptions` in the `utils/` directory and iterate over them.

Reply:
Cool, you have implemented it in the same way as `countryOptions` were already present.

Reply:
Hey @palisadoes, I'm considering making a PR to implement the **enums** on Talawa-Admin. Since these enums will be widely used and having separate PRs for each component might lead to conflicts, I'm thinking of creating a single PR for both the enums and the dropdown component. This would streamline the process and reduce potential conflicts. What do you think? Can I proceed with this approach?

Reply:
Hey @pranshugupta54, I have implemented the changes and ensured that translations for the labels are done wherever necessary. I believe it's crucial for us to utilize the multilingual support available in our application.